savant/deepstream/source_output.py

```diff
@@ -421,7 +421,14 @@ def __init__(
         )
         self._output_frame = output_frame
         self._encoder = self._codec.encoder(output_frame.get('encoder'))
-        self._params = output_frame.get('encoder_params') or {}
+
+        # check if {codec}_encoder_params are available
+        encoder_params_key_name = f'{self._codec.name.lower()}_encoder_params'
+        encoder_params = output_frame.get(encoder_params_key_name, {})
+        if not encoder_params:
+            encoder_params = output_frame.get('encoder_params', {})
+
+        self._params = encoder_params or {}
```

Comment:
I think you can replace it with one instruction `self._params = output_frame.get(encoder_params_key_name, {}) or output_frame.get('encoder_params', {})`

In any case you don't need `or {}` here, because you have defaults {}